src/features/FindReferencesManager.js

```diff
@@ -0,0 +1,140 @@
+        SearchModel                 = require("search/SearchModel").SearchModel,
+        Strings                     = require("strings");
+
+    var _providerRegistrationHandler = new ProviderRegistrationHandler(),
+        registerFindReferencesProvider = _providerRegistrationHandler.registerProvider.bind(
+            _providerRegistrationHandler
+        ),
+        removeFindReferencesProvider = _providerRegistrationHandler.removeProvider.bind(_providerRegistrationHandler);
+
+    var searchModel = new SearchModel(),
+        _resultsView;
+
+    function _getReferences(provider, hostEditor, pos) {
+        var result = new $.Deferred();
+
+        if(!provider) {
+            return result.reject();
+        }
+
+        provider.getReferences(hostEditor, pos)
+            .done(function (rcvdObj) {
+
+                searchModel.results = rcvdObj.results;
+                searchModel.numFiles = rcvdObj.numFiles;
+                searchModel.numMatches = rcvdObj.numMatches;
+                searchModel.allResultsAvailable = true;
+                searchModel.setQueryInfo({query: rcvdObj.queryInfo, caseSensitive: true, isRegExp: false});
+                result.resolve();
+            }).fail(function (){
+                result.reject();
+            });
+        return result.promise();
+
+    }
+
+    function _openReferencesPanel() {
+        var editor = EditorManager.getActiveEditor(),
+            pos = editor ? editor.getCursorPos() : null,
+            referencesPromise,
+            result = new $.Deferred(),
+            errorMsg = Strings.REFERENCES_NO_RESULTS,
+            referencesProvider;
+
+        var language = editor.getLanguageForSelection(),
+            enabledProviders = _providerRegistrationHandler.getProvidersForLanguageId(language.getId());
+
+        enabledProviders.some(function (item, index) {
+            if (item.provider.hasReferences(editor)) {
+                referencesProvider = item.provider;
+                return true;
+            }
+        });
+
+        referencesPromise = _getReferences(referencesProvider, editor, pos);
+
+        // If one of them will provide a widget, show it inline once ready
+        if (referencesPromise) {
+            referencesPromise.done(function () {
+                _resultsView.open();
+            }).fail(function () {
+                _resultsView.close();
+                editor.displayErrorMessageAtCursor(errorMsg);
+                result.reject();
+            });
+        } else {
+            _resultsView.close();
+            editor.displayErrorMessageAtCursor(errorMsg);
+            result.reject();
+        }
+
+        return result.promise();
+    }
+
+    /**
+     * @private
+     * Clears any previous search information, removing update listeners and clearing the model.
+     */
+    function _clearSearch() {
+        searchModel.clear();
+    }
+
+    AppInit.appReady(function () {
```

Comment:
This should be **htmlReady**, I think. For non-LSP based providers, DOM initialization is more than enough. See my comment above.